### PR TITLE
thermistor: add support for thermistors such as the NTC 3950

### PR DESCRIPTION
**Pull request by deadprogram**
This PR add support for thermistors such as the NTC 3950.

### CODE REVIEW

File: thermistor/thermistor.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	}
+}
+
+// Configure configures the ADC pin used for the thermistor.
+func (d *Device) Configure() {
+	d.adc.Configure()
+}
+
+// Temperature returns the temperature in celsius milli degrees (ºC/1000)
+func (d *Device) Temperature() (temperature int32, err error) {
```
Comment (aykevl) [inline thermistor/thermistor.go]: See https://github.com/tinygo-org/drivers/pull/38#discussion_r277324240

Comment (deadprogram) [reply thermistor/thermistor.go]: OK, let's go with ReadTemperature() (int32, error) across all drivers. Change made.

**Comment by deadprogram** [comment]
This PR should now be ready for review.

**Comment by deadprogram** [comment]
Added smoke test to PR.

**Comment by aykevl** [comment]
The only thing I'm not really happy about is the use of floats. But I guess converting this to integer math with the same amount of precision is difficult, so maybe we should keep it this way.
Feel free to update or merge.

**Pull request merged**